cmd/kubeadm/app/util/output/output.go

```diff
@@ -0,0 +1,201 @@
+	KubeadmTextPrintFlags  KubeadmTextPrintFlags
+
+	TypeSetterPrinter *printers.TypeSetterPrinter
+
+	OutputFormat *string
+
+	// OutputFlagSpecified indicates whether the user specifically requested a certain kind of output.
+	// Using this function allows a sophisticated caller to change the flag binding logic if they so desire.
+	OutputFlagSpecified func() bool
+}
+
+// AllowedFormats returns list of allowed output formats
+func (kof *KubeadmOutputFlags) AllowedFormats() []string {
+	ret := []string{"text"}
+
+	ret = append(ret, kof.JSONYamlPrintFlags.AllowedFormats()...)
+	ret = append(ret, kof.KubeTemplatePrintFlags.AllowedFormats()...)
+
+	return ret
+}
+
+// ToPrinter receives an outputFormat and returns a printer capable of
+// handling format printing.
+// Returns error if the specified outputFormat does not match supported formats.
+func (kof *KubeadmOutputFlags) ToPrinter() (KubeadmPrinter, error) {
+	outputFormat := ""
+	if kof.OutputFormat != nil {
+		outputFormat = *kof.OutputFormat
+	}
+
+	if kof.KubeadmTextPrintFlags != nil {
+		if p, err := kof.KubeadmTextPrintFlags.ToPrinter(outputFormat); !genericclioptions.IsNoCompatiblePrinterError(err) {
+			return p, err
+		}
+	}
+
+	if kof.JSONYamlPrintFlags != nil {
+		if p, err := kof.JSONYamlPrintFlags.ToPrinter(outputFormat); !genericclioptions.IsNoCompatiblePrinterError(err) {
+			return NewResourcePrinterWrapper(kof.TypeSetterPrinter.WrapToPrinter(p, err))
+		}
+	}
+
+	if kof.KubeTemplatePrintFlags != nil {
+		if p, err := kof.KubeTemplatePrintFlags.ToPrinter(outputFormat); !genericclioptions.IsNoCompatiblePrinterError(err) {
+			return NewResourcePrinterWrapper(kof.TypeSetterPrinter.WrapToPrinter(p, err))
+		}
+	}
+
+	return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: kof.OutputFormat, AllowedFormats: kof.AllowedFormats()}
+}
+
+// AddFlags receives a *cobra.Command reference and binds
+// flags related to Kubeadm printing to it
+func (kof *KubeadmOutputFlags) AddFlags(cmd *cobra.Command) {
+	kof.JSONYamlPrintFlags.AddFlags(cmd)
+	kof.KubeTemplatePrintFlags.AddFlags(cmd)
+	cmd.Flags().StringVarP(kof.OutputFormat, "experimental-output", "o", *kof.OutputFormat, fmt.Sprintf("Output format. One of: %s.", strings.Join(kof.AllowedFormats(), "|")))
+}
+
+// WithDefaultOutput sets a default output format if one is not provided through a flag value
+func (kof *KubeadmOutputFlags) WithDefaultOutput(outputFormat string) *KubeadmOutputFlags {
+	kof.OutputFormat = &outputFormat
+	return kof
+}
+
+// WithTypeSetter sets a wrapper than will surround the returned printer with a printer to type resources
+func (kof *KubeadmOutputFlags) WithTypeSetter(scheme *runtime.Scheme) *KubeadmOutputFlags {
+	kof.TypeSetterPrinter = printers.NewTypeSetter(scheme)
+	return kof
+}
+
+// NewKubeadmOutputFlags creates new KubeadmOutputFlags
+func NewKubeadmOutputFlags(kubeadmTextPrintFlags KubeadmTextPrintFlags) *KubeadmOutputFlags {
```

Comment:
this should accept a pointer to `KubeadmTextPrintFlags`.

Reply:
done